src/instamatic/gui/ctrl_frame.py

```diff
@@ -31,59 +34,47 @@ def __init__(self, parent):
         b_find_eucentric_height = Button(
             frame, text='Find eucentric height', command=self.find_eucentric_height
         )
-        b_find_eucentric_height.grid(row=0, column=0, sticky='EW', columnspan=2)
+        b_find_eucentric_height.grid(row=0, column=0, sticky='EW')
 
         Label(frame, text='Mode:').grid(row=8, column=0, sticky='W')
-        self.o_mode = OptionMenu(
-            frame,
-            self.var_mode,
-            'diff',
-            'diff',
-            'mag1',
-            'mag2',
-            'lowmag',
-            'samag',
-            command=self.set_mode,
-        )
-        self.o_mode.grid(row=8, column=1, sticky='W', padx=10)
+        ranges = config.microscope.ranges
+        modes = list(ranges.keys())
```

Comment:
Python always iterates over the dict keys, so it's not necessary to call keys.

```suggestion
        modes = list(config.microscope.ranges)
```

Reply:
The panels gets only as much love as time I can spare now, unfortunately.
Also, as `import this` verse 2 proclaimeth, *Explicit is better than implicit*